src/stock-receipt/stock-receipt.jsx

```diff
@@ -0,0 +1,293 @@
+import React, { useEffect, useState } from "react";
```

Comment:
There is a lot of code in the .jsx file (Frontend code + business logic of parsing response + Qty calculation, etc). 

It would be clearer separation of concerns if the .jsx file ONLY did frontend manipulation, and all the business logic of talking to eAushadya and mapping was in a different JS file. 

For instance, the UI is just table-view & stock-data. When an outward number is entered, the number is sent to an eAushadya adapter, which makes the REST call, gets the response, and hands the whole respone to a Mapper. The Mapper job is to parse the respose, and construct the "Stock-Data" for Frontend to render. Then the mapper returns the "Stock-Data" and frontend just renders it as-is. This allows for Frontend react code to be totally disconnected from eAushadya.

Reactcode -> AushdayaAPIcaller -> Mapper -> ReactCode.

This way Mapper is easy to test.
Also, the code which computes the Qty is business logic for EAushadya. Mapper can do that job.

Reply:
changes have been made. Moved business logic to helper file

Reply:
A helper file is usually a code smell (esp if it contains business logic). Every code file is a "helper" : )
What help does it provide? That should be the name of the class/file.

In this case, does it perform Row->Stock Transation mapping, then call it stock-mapper. Or if it performs Aushadya response mapping to Stock Row, then call it AushadyaResponseMapper, etc. 

If this file is not precisely named, then all business logic will keep going into "helper" file in the future, because it is generically named.

Reply:
moved logic near to stock-receipt.jsx in the file name eaushadha-response-mapper.js